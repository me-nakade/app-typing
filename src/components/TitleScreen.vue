<template>
  <div class="title-screen">
    <h1>タイピングゲーム</h1>
    <ul class="instructions">
      <li>表示される文字列と同じように入力しましょう(制限時間あり！)</li>
      <li>「。」や「.」は不要です</li>
      <li>入力し終えたら、力強くエンターキーを押しましょう</li>
    </ul>
  </div>
  <button @click="$emit('start')">スタート</button>
</template>

<style scoped>
.title-screen {
  text-align: center;
  margin-top: 100px;
}

button {
  font-size: 2em;
  padding: 0.5em 2em;
  margin: 1em auto;
  display: block;
}

.instructions {
  display: inline-block;
  padding: 0;
  list-style-position: inside;
}

.instructions li {
  text-align: left;
}
</style>
