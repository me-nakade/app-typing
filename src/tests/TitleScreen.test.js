import { describe, it, expect } from 'vitest'
import { mount } from '@vue/test-utils'
import TitleScreen from '../components/TitleScreen.vue'

describe('TitleScreen.vue', () => {
  it('タイトルが表示される', () => {
    const wrapper = mount(TitleScreen)
    expect(wrapper.find('h1').text()).toBe('タイピングゲーム')
  })

  it('説明リストが3つ表示される', () => {
    const wrapper = mount(TitleScreen)
    const items = wrapper.findAll('.instructions li')
    expect(items.length).toBe(3)
    expect(items[0].text()).toContain('表示される文字列と同じように入力しましょう')
    expect(items[1].text()).toContain('「。」や「.」は不要です')
    expect(items[2].text()).toContain('入力し終えたら、力強くエンターキーを押しましょう')
  })

  it('スタートボタンが表示される', () => {
    const wrapper = mount(TitleScreen)
    const button = wrapper.find('button')
    expect(button.exists()).toBe(true)
    expect(button.text()).toBe('スタート')
  })

  it('スタートボタンをクリックするとstartイベントがemitされる', async () => {
    const wrapper = mount(TitleScreen)
    await wrapper.find('button').trigger('click')
    expect(wrapper.emitted('start')).toBeTruthy()
    expect(wrapper.emitted('start').length).toBe(1)
  })

  it('スナップショットが一致する', () => {
    const wrapper = mount(TitleScreen)
    expect(wrapper.html()).toMatchSnapshot()
  })
})
