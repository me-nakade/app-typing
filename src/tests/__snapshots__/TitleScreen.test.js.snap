// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TitleScreen.vue > スナップショットが一致する 1`] = `
"<div data-v-cb2ce24b="" class="title-screen">
  <h1 data-v-cb2ce24b="">タイピングゲーム</h1>
  <ul data-v-cb2ce24b="" class="instructions">
    <li data-v-cb2ce24b="">表示される文字列と同じように入力しましょう(制限時間あり！)</li>
    <li data-v-cb2ce24b="">「。」や「.」は不要です</li>
    <li data-v-cb2ce24b="">入力し終えたら、力強くエンターキーを押しましょう</li>
  </ul>
</div>
<button data-v-cb2ce24b="">スタート</button>"
`;
